csrc/id_model/loop_promotion.h

```diff
@@ -14,6 +14,78 @@ namespace nvfuser {
 class IdModel;
 struct StatefulInliningInfo;
 
+struct CoveredGroup;
+
+using CoveredGroups = std::unordered_set<CoveredGroup>;
+
+std::string toString(const CoveredGroups& covered_groups);
+
+// Represents an input (or split output) ID group that an exact group
+// depends on (i.e., covers). If an input ID group is split, split_in_
+// refers to the covered groups of the input ID group, and group_
+// refers to either inner or outer output group.
+struct CoveredGroup {
```

Comment:
Does `CoveredGroup` defines a language to say things like: the inner of the outer of the inner of x, y and z?